app/views/search/index.html.erb

```diff
@@ -18,41 +17,58 @@
 </div>
 
 <%= form_tag(search_path, method: :get) do %>
-    <div class="input-group mb-3">
-        <%= text_field_tag :search, params[:search], class: "form-control" %>
+    <div class="input-group mb-3 mt-3">
+        <%= text_field_tag :search, params[:search], class: "form-control", placeholder: "Suche" %>
         <button type="submit" class="btn btn-outline-secondary">
             <i class="bi bi-search"></i>
         </button>
         <button type="button" class="btn btn-outline-secondary" data-bs-toggle="modal" data-bs-target="#qrscanner">
             <i class="bi bi-qr-code-scan"></i>
         </button>
     </div>
-    <div class="btn-group mb-3">
-        <button class="btn btn-secondary dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false">
-            Filter
-        </button>
-        <div class="dropdown-menu">
-            <div class="input-group">
-                <%= select_tag "filter_property", raw("<option>Title</option><option>Author</option><option>Release</option>"), class: "form-select" %> <%# https://apidock.com/rails/ActionView/Helpers/FormTagHelper/select_tag %>
-                <%= text_field_tag :filter_value, params[:filter_value], class: "form-control" %>
+    
+    <button class="btn btn-secondary" type="button" data-bs-toggle="modal" aria-expanded="false" data-bs-target="#filter-modal">
+        FILTER <i class="bi bi-sliders2-vertical"></i>
+    </button>
+    
+    <div class="modal-fullscreen modal fade" id="filter-modal" tabindex="-1" aria-labelledby="filter-modal-title" aria-hidden="true">
+        <div class="modal-dialog">
+            <div class="modal-content">
+                <div class="modal-header">
+                    <h1 class="modal-title fs-5" id="filter-modal-title">Modal title</h1>
+                    <button type="button" class="btn-close" data-bs-dismiss="modal" aria-label="Close"></button>
+                </div>
+                <div class="modal-body">
+                    <div class="input-group">
+                        <%= select_tag "filter_property", raw("<option>Title</option><option>Author</option><option>Release</option>"), class: "form-select" %> <%# https://apidock.com/rails/ActionView/Helpers/FormTagHelper/select_tag %>
+                        <%= text_field_tag :filter_value, params[:filter_value], class: "form-control" %>
+                    </div>
+                </div>
+                <div class="modal-footer">
+                    <button type="button" class="btn btn-secondary" data-bs-dismiss="modal">Close</button>
+                    <button type="button" class="btn btn-primary">Save changes</button>
+                </div>
             </div>
         </div>
     </div>
```

Comment:
Do reduce the code of the view, I would recommend to put this modal into it's own partial. What do you think?

Reply:
We moved both modals to partials